spec/postgres_url_spec.cr

```diff
@@ -0,0 +1,21 @@
+require "./spec_helper"
+
+describe Avram::PostgresURL do
+  it "strips space and newlines from the username" do
+    creds = Avram::PostgresURL.build(database: "test", username: " oops\n")
+
+    creds.username.should eq "oops"
+  end
+
+  it "raises an InvalidDatabaseNameError when the database name is blank" do
+    expect_raises(Avram::InvalidDatabaseNameError) do
+      Avram::PostgresURL.parse("")
+    end
+  end
+
+  describe "parse" do
+    it "returns nil when there's nothing to parse" do
+      Avram::PostgresURL.parse(nil).should eq nil
+    end
+  end
+end
```

Comment:
Would it worth testing the unix command thing while you're in here considering that's what kicked off the whole refactor?

Reply:
Well, this thing doesn't actually do any sort of connecting. Just building of the string. I could test that a unix socket style string is formatted. Like `postgres:///test_db`

Reply:
Mmm fair. Up to you.